source/plugin/practices.rst

```diff
@@ -119,3 +128,68 @@ Plugin Interoperability
 =======================
 
 An explanation of how to communicate with other plugins, *TBA*.
+
+
+Bad Practices
+=============
+
+These should practices should be avoided, as they can lead to memory leaks (``OutOfMemoryError``), lag or
+inconsistencies.
+
+
+Storing References
+~~~~~~~~~~~~~~~~~~
+
+Some instances such as 
+
+* :javadoc:`DataHolder`
+* :javadoc:`TileEntity`
+* :javadoc:`CommandSource`
+* :javadoc:`Entity`
+* :javadoc:`Player`
+* :javadoc:`Subject`
+* :javadoc:`World`
+* and containers that **MIGHT** contain any of the above elements, including
+    * ``Collections``
+    * ``Maps``
+
+should **NEVER** be stored or cached in plugins.
+
+These are the main reasons for this:
+
+* The references prevent proper garbage collection
+* The instances might no longer be valid
+
+This can easily be avoided by using the corresponding snapshots or saving the UUID of the given instances and requesting
+a live instance when you need it.
+
+
+IO on the main thread
+~~~~~~~~~~~~~~~~~~~~~
+
+Executing some IO operations such as loading a config/data file or checking for updates/connecting to a website takes
+much time and greatly affects the TPS on the server. Such tasks should be done either in their own threads, or using the
+inbuilt scheduler's async feature. It is perfecty fine to load required files/config on the main thread during server
+startup/plugin initialization though.
+
+.. code-block:: text
+
+    this.game.getScheduler().createAsyncExecutor(this).execute(this::checkForUpdates);
+
+For more details refer to the :doc:`scheduler` docs.
+
+If this is done wrong, you will see clients timing out or the server's watchdog might kill the server.
```

Comment:
I suggest "If this is done incorrectly, you may see clients time out, or the watchdog may even kill the server."